superset/charts/api.py

```diff
@@ -465,13 +465,16 @@ def data(self) -> Response:
             return self.response_400(message="Request is incorrect")
         except ValidationError as error:
             return self.response_400(
-                _("Request is incorrect: %(error)s", error=error.messages)
+                message=_("Request is incorrect: %(error)s", error=error.messages)
             )
         try:
             query_context.raise_for_access()
         except SupersetSecurityException:
             return self.response_401()
         payload = query_context.get_payload()
+        for query in payload:
+            if query["error"]:
+                return self.response_400(message=f"Error: {query['error']}")
```

Comment:
Would be great if `query_context.get_payload()` would raise custom superset exceptions instead of checking for errors on the payload. But we can address that later

Reply:
That's a great idea 👍  To unblock `0.37` I'll merge this as-is, but will improve exception handling after we ship the release.